Assorted fixes and improvements

WolframRavenwolf opened the pull request:
Hey, it's been a while, but I've got a bunch of stuff done and more planned:

**DONE** (in this PR):

- Kobold Presets fixed:
  - Opened every preset and saved it, then removed the cross-preset settings ("streaming_kobold", "genamt", "max_length") to ensure every preset has the correct order and contents.
  - "Ban EOS Token" should be off since all modern models are finetuned to output EOS tokens and ignoring those will derail the chat by going "out of bounds" when forcing them to keep generating after response is considered complete by the model.
    - Removed `use_default_badwordsids` from kobold presets
    - ~~Took "Ban EOS Token" and "Single-line mode" out of presets and moved them into the new global section above preset selector.~~
    - Removed "single_line" from kobold presets
    - Removed obsolete "use_stop_sequence" setting
    - "Ban EOS Token" off by default
    - ~~Split AI Resp. Conf. in global and preset-specific settings~~
- Help texts fixed:
  - Unlocked Context Size: "Only select models support context sizes greater than 2048 tokens. Increase only if you know what you're doing." -> "Only select models support context sizes greater than 4096 tokens. Increase only if you know what you're doing." (Slider already goes up to 4096 tokens when locked, but the help text hadn't been updated, so I fixed that now.)
  - Character Handling: "Prefer Char. JB" -> "Prefer Char. Jailbreak" (clearer and it's not too long)
  - /sendas: "<br>Example:<br>" -> " Example:" (whitespace fix, as `br` is ignored and `\n` becomes space)
  - /world: Added "optional name" argument, adhering to standard format
  - /api: Added OpenRouter and Scale, and used monospace for args, adhering to standard format
  - /impersonate, /delchat, /closechat, /panels, /qr, /qrset: adhering to standard format with dash instead of minus
  - /newchat, /random: alias is same as command, so removed redundant aliases
  - /cut: Removed misspelled "requred", adhering to standard format which doesn't specify required for numbers (like /qr, /depth, /freq)
  - /sprite, /spriteoverride: adhering to standard format with args in brackets, and spriteoverride optional folder
  - /show-gallery: "Shows the gallery" -> "– shows the gallery" (adhering to standard format)
  - /qr: "- requires number argument," -> "(number) –" (adhering to standard format)
  - /qrset: "- arg: QuickReply Preset Name, swaps to that QR preset" -> "– swaps to the specified Quick Reply Preset" (adhering to standard format and spelling)
  - Changed "aliases" to "alias" since all commands only have one alias so far, and this shortens the help text a little
  - Removed unnecessary whitespace at beginning of slash command help strings
  - Fix title
  - Add translations for new help texts
  - Fix i18n.json whitespace (proper indentation, spaces instead of tabs)
- Scale API for connectAPISlash:
  - /api: Added Scale API to `connectAPISlash` function
- Quick Reply checkboxes fixed:
  - Quick Reply checkboxes were too far apart, put them closer together, like all the other extensions' checkboxes
- New Instruct Mode Presets:
  - ChatML (OpenAI's Chat Markup Language-inspired) context template and instruct prompt format, used by e. g. [Open-Orca/Mistral-7B-OpenOrca · Hugging Face](https://huggingface.co/Open-Orca/Mistral-7B-OpenOrca#prompt-template)
    - Renamed "ChatML" to "Mistral-OpenOrca (ChatML)" to make it clearer what this is for.
    - Renamed "Mistral-OpenOrca" (removed "ChatML" from its name).
    - Renamed "Mistral-OpenOrca" to "ChatML" again (for lack of a better name).
  - Libra-32B: [Envoid/Libra-32B · Hugging Face](https://huggingface.co/Envoid/Libra-32B#update-silly-tavern-format)
  - Mistral: [mistralai/Mistral-7B-v0.1 · Hugging Face](https://huggingface.co/mistralai/Mistral-7B-v0.1)
  - Synthia: [migtissera/Synthia-7B-v1.3 · Hugging Face](https://huggingface.co/migtissera/Synthia-7B-v1.3#here-is-prompt-format)
- More date/time macros:
  - {{weekday}} - the current weekday
  - {{isotime}} - the current ISO date (YYYY-MM-DD)
  - {{isodate}} - the current ISO time (24-hour clock)
  - {{datetimeformat …}} - the current date/time in the specified format, e. g. for German date/time: {{datetimeformat DD.MM.YYYY HH:mm}}
- More (UI) fixes and improvements
  - Reverted some changes (crossed them out in the above changelog)
  - Renamed Kobold's "Rep. Pen. Slope" to "Repetition Penalty Slope" and moved it directly beneath the other Rep. Pen. settings
  - Renamed TextGen's "Encoder Rep. Pen." to "Encoder Repetition Penalty" for consistency (string lenght is fine)
  - Unhid "Repetition Penalty Range" in Simple mode because it's highly relevant when Context Size (which is visible in Simple mode) is increased
  - Removed "single_line" and "use_default_badwordsids" from Preset Manager filteredKeys again for consistency (and their defaults are fine now)

**TODO** (I can add it to this PR or create a new one or more):

- Message sound only on final generation when doing multi-gen/auto-continue
- Model detection on generation instead of connection when local (I often load a new model and the change is not reflected in the metadata when I forget to click Connect again)
- More advanced date macros:
  - Is date/time localization possible? E. g. when using a German model, dates should be in German format.
  - Date of/time since last session with char (requires checking if a previous chat log exists and figuring out the date/time of its last message - seems quite complicated but would allow characters to act as if they remembered how much time passed since the last chat, or even realize this is the first time they talk to user, making them act much more believably)
- Quick reply regex for automatic preset selection based on char
- Auto-save Message Edits should apply even when disabled when sending while editing
- Extensions should be listed in alphabetical instead of random, changing order
- There are multiple mentions of "Typical Sampling" and "Typical P Sampling". Shouldn't we consistently use one of the two terms? If so, which one?
- Should we reorder settings for consistency? In KoboldAI settings, the order according to sampler IDs would be Top K, A, P, TFS, Typical instead of current order P, A, K, Typical.
- BUGFIX Auto-Scroll chat "sticky" not working on PC
- BUGFIX Voice Recognition on mobile (tested on Chrome on Android: only first word is returned)
- BUGFIX Auto-Continue causing regen loop if the first few tokens are stopping strings
- BUGFIX User Messages without AI response in between messes up Sequences (but are consecutive User messages even supported/normal?!)
- BUGFIX Example messages wrapped in Sequences can break complicated prompt formats like Llama 2 Chat's
- BUGFIX When swiping greeting message after chat was started and messages were deleted, username variable is gone
- BUGFIX Stopping string not always removed, e. g. when following directly after a word (example: "xyz.title:" with "title:" a custom stopping string)
- BUGFIX When entering number in Context Size while Unlocked is OFF, upon reload, the value is trimmed. Instead, IMHO, it should enable Unlocked and accept the manually entered and displayed value.
- BUGFIX: All ooba/Mancer settings have fractional units, even the integer ones! (Caused by https://github.com/SillyTavern/SillyTavern/blob/staging/public/scripts/textgen-settings.js#L285)

WolframRavenwolf committed:
Kobold Presets fixed

WolframRavenwolf committed:
Help texts fixed

WolframRavenwolf committed:
Scale API for connectAPISlash

WolframRavenwolf committed:
Quick Reply checkboxes fixed

WolframRavenwolf committed:
New Instruct Mode Presets

Cohee1207 commented:
Can't agree to the Ban EOS thing, this is going to be a massive controversy, you can quote me on that. Are there any proofs or tests for that choice?

WolframRavenwolf commented:
> Can't agree to the Ban EOS thing, this is going to be a massive controversy, you can quote me on that. Are there any proofs or tests for that choice?

Why do you think that? **Un**banning the EOS token should have been the default for a long time, and it already is for llama.cpp (which koboldcpp is based on) as well as oobabooga's text-generation-webui, they all accept the EOS token by default to end inference when the model thinks it's done (instead of forcing it to continue, going out of bounds, derailing the chat).

[The KoboldCpp FAQ and Knowledgebase](https://github.com/LostRuins/koboldcpp/wiki#koboldcpp-general-usage-and-troubleshooting) talks about it under "My AI continues rambling / writing rambling after it should have stopped generating / What does Unban Tokens do?", but unfortunately doesn't stress enough how important it is for coherent responses. Modern models are finetuned with EOS tokens, and not respecting them causes them to hallucinate and ramble on, causing issues that are hard to troubleshoot for people who don't know these finer details of inference.

Users might think the model or software is faulty because it generates nonsense, or they might not even notice that they're getting subpar results. The only situation where ignoring the EOS token is useful is when you want to generate very long stories without user interaction, so you let it fill up the context on its own. Anything that's a back and forth between the AI and the user, i. e. a conversation, benefits from the EOS token. So I've continuously been using and recommending this option.

And just think about it: What other option do we have to end inference before the max new token limit is hit? Stopping strings! We try to catch the model talking as the user and stop inference when it does. However, while the EOS token is never part of the input, the user's name most likely is. And as the conversation goes on, repetition penalty will affect it, making the stopping string less likely to appear correctly, causing messages to get longer and less coherent until the conversation eventually derails completely.

All those issues can be avoided by respecting the EOS token. That's why I push for it to be the (only sensible) default.

**Edit:** Also somewhat related: https://github.com/LostRuins/koboldcpp/issues/305#issuecomment-1629796932 - the old sampler order problem caused the EOS token to be suppressed and the outcome is perfectly illustrated by the examples in that comment.

Cohee1207 commented:
What are the model finetunes in question? It's important to remember that the presets are going to apply to all models, both old and new. Please raise this to all parties involved in Kobold-alikes development (not only CPP, United is still widely used - @henk717). If they make the unban their default state, we can make this change in ST accordingly.

https://github.com/henk717/KoboldAI/blob/united/aiserver.py#L8180

WolframRavenwolf committed:
More date/time macros

WolframRavenwolf commented:
> What are the model finetunes in question? It's important to remember that the presets are going to apply to all models, both old and new. Please raise this to all parties involved in Kobold-alikes development (not only CPP, United is still widely used - @henk717). If they make the unban their default state, we can make this change in ST accordingly.
> 
> https://github.com/henk717/KoboldAI/blob/united/aiserver.py#L8180

Pretty much all finetunes of LLaMA (1) and Llama 2 models should be using EOS tokens properly. It's part of the official Llama generation reference implementation: [llama/llama/generation.py at main · facebookresearch/llama](https://github.com/facebookresearch/llama/blob/main/llama/generation.py) - stopping inference any other way, e. g. using stopping strings, is just a workaround since our models aren't yet smart enough to always end their messages properly and often keep going anyway, hallucinating the user's next message.

The code you linked refers to "badwordsids", "tokens that commonly worsen the writing experience for continuous story writing". I wonder if that's legacy code since I'd not call the EOS token a "bad word", maybe there were or are other tokens that they refer to? But the description is clear, it's about continuous story writing, something I'd not consider the primary use case for SillyTavern since the interface is more suited to interactive conversations than continuous story-writing, wouldn't you agree? So IMHO, we should make chat work as well as possible by default, while continuous story-writing would still be a possibility, and users of that could just ban EOS token again (just like they would have to alter other settings for optimal story-writing, like new token limits and creating/downloading storyteller characters, etc.).

Anyway, I opened this discussion: [Please Unban EOS Token by default · LostRuins/koboldcpp · Discussion #459](https://github.com/LostRuins/koboldcpp/discussions/459) and the dev said "I agree that allowing EOS as the default if not specified in the API payload does make sense." so the default might change for koboldcpp. What's more interesting, though: The KoboldLite UI has found an even better solution than a binary toggle, and maybe we could implement this for SillyTavern as well:

[auto EOS mode, save streaming setting · LostRuins/lite.koboldai.net@03723d0](https://github.com/LostRuins/lite.koboldai.net/commit/03723d0d3dd69a3adb7b6b00b3674b2e7ead0378)

Instead of an on/off switch, EOS Token Ban can also be set to "Auto", and that's the new default. With that, when you send a message, the EOS token is dynamically enabled or disabled based on certain conditions:

- In Story Mode, the EOS token is always banned.
- In Chat Mode, the EOS token is generally allowed.
- But if Auto-Continue is used and the last AI reply was empty, then the EOS token is banned for the next reply.

This is super clever, because there's one thing with the EOS token: Models tend to be very certain that their reply is finished, so when you click continue, they will simply send the EOS token again. This is a good thing when using Auto-Continue like I do, because it ensures you get the full response, nothing more and nothing less. But in rare cases, it's possible that the model outputs the EOS token but you might still want it to keep going, to generate more. So automatically banning it temporarily to get a reply after you got an empty one (i. e. it generated just the EOS token and the reply is empty because of that) could be a useful feature.

However, considering that this isn't as trivial as I initially thought, maybe it's best to revert the EOS setting change and do an independent PR for this topic? I'd remove the EOS setting from the presets, just like the streaming setting, so it's not (un)set by changing presets. At the same time, I'd like to implement a slight UI optimization:

Since we have several settings that are on the presets page, but not part of the presets (response length, context size, streaming, ban EOS token, possibly grammar), what do you say about properly separating the global and preset-specific settings by moving the preset selector in between them? Anything above the selector isn't part of the presets, so doesn't change when you select one, while anything below the selector is overwritten by the selection. I could implement that as I revert the EOS change.

IkariDevGIT commented:
@StefanDanielSchwarz i think setting EOS token to banned(or not) automatically is not a good idea, i agree that it should not be banned by default tho.

WolframRavenwolf committed:
ChatML context template and instruct prompt format

WolframRavenwolf committed:
Mistral context template and instruct prompt format

WolframRavenwolf committed:
Removed use_default_badwordsids from kobold presets

WolframRavenwolf commented:
@Cohee1207 OK, I've removed `use_default_badwordsids` again from all KoboldAI Settings presets!

I've also started to experiment with the AI Response Configuration UI - what do you think about this:

![AI Response Configuration](https://github.com/SillyTavern/SillyTavern/assets/52386626/fade2254-b151-4535-8c5f-cb01efe1c428)

My goal is to make it clear which settings are global and which are preset-specific. So everything above the preset selector is global, and everything below it is saved and loaded with the presets.

Also added a help text to "Ban EOS Token". And "Single-line mode" is global, too, as I think it's a global thing as well (if it isn't, I'll move it back down). What about Grammar, should that be preset-specific or global?

henk717 commented:
Chiming in here since there is a small difference in Koboldcpp and KoboldAI United for legacy reasons.

use_default_badwordsids will automatically turn itself back on in United if not explicitly set to false in the API request. I did this because of legacy reasons and to avoid sudden changes in the API's behavior (Mostly for Horde in the transition period).

It does slightly more on United than it does on Koboldcpp, concedo decided not to adopt the same behavior so in Koboldcpp this is being used as a toggle for the EOS.

In United it is customizable by the model developer which has made it less trivial for me to add the settings. Its not just EOS, we automatically generate a list that bans [ ] on top of the EOS token so users can use it in ways that prevent World Info leaking. The philosophy is that UI's handle their own stop's with stop sequence requests so that irs the UI that decides when to stop and not the model.

But for newer instruct and especially chat models this can give issues for models who were trained with very frequent EOS use instead of multi turn instruct formatting like in my opinion they should have. So our ban the EOS philosophy while essential for story writing is not always be the ideal setting for chat models and heavily depends on what you are using.

In an ideal world all models use our config settings to customize the badwordsids list as they desire, Pygmalion for example does this. But with how big the tuning space has gotten this is not realistic.

On the other side of the issue is that customizability we grant model creators. Its a tricky one since I do not know in advance what tokens are being banned with the Badwordsids ban in case models customize them. For that reason its not spread out to multiple settings as that would have broken the old configs for our model partners.

I don't rule out that I find a way to still add a seperate unban eos option to the API but for now I don't have a good method for that.

Which default to use is a bit trickier because of that, I do see most of our ST users preferring the EOS unbanned so I do think its a default that makes sense for you. But that does come at the price of [ ] leaking (Which on KCPP can happen anyway). You could implement a seperate regex for this that strips everything between those brackets from the end result, if you do that for a chat UI having the EOS unbanned by default is a nobrainer.

For KoboldAI Lite we have indeed shifted behavior, in story mode we ban it by default since story writers need that EOS banned or they get constant refusals. While in the other models we turn it on when the model had an empty generation so our users still have a way to continue but it also doesn't overflow immediately.

WolframRavenwolf commented:
Thanks for chiming in, @henk717, that's very insightful information.

So that's why it's called "badwordsids" in the background. Considering that it bans different things in different backends (only EOS token in koboldcpp, but also square brackets and possibly others as defined by the model in KoboldAI), and in accordance with your statement that "most of our ST users preferring the EOS unbanned", I think we should have "Ban EOS Token" OFF by default in SillyTavern.

That is what other frontends and backends do as well and the least confusing, IMHO, as it gives the unadulterated output the model delivers. If anything like [ ] leaking happens, the user sees it and can act accordingly (e. g. add the square brackets to stopping strings or ban them or set a regex, just like they would for any other kind of unwanted output).

@Cohee1207 What do you think about this now? How about presenting it like this:

![Ban EOS Token Helptext](https://github.com/SillyTavern/SillyTavern/assets/52386626/38d67a57-d5b4-4185-99dd-5ca917280bec)

I think the help text as a mouse-over would be better than a fixed text underneath the option to prevent the UI getting stretched so much. If you're OK with it, I'll implement it that way.

WolframRavenwolf committed:
Renamed ChatML to Mistral-OpenOrca (ChatML)

Cohee1207 commented:
Singleline seems like a legacy thing too, Koboldcpp might not support it at all. It's very easily emulated however by adding a newline as a custom stopping string on all backends, that's how it essentially works.

Leaving EOS out of presets is a good option. A user can make an informed decision to turn it on or off and leave it as they prefer, based on the hint text or their own judgment. To exclude it from custom, not built-in presets, it needs to be added to the ignore list in preset-manager.js, so it won't overwrite the chosen value on preset picking. About the default value for new installs, we may want to experiment with setting it to false (**UN**ban the EOS). Can be rolled back if any negative feedback occurs.

Mouse-over text is not an option for mobile users (we need to care about them too), so grayed-out small text is okay for me.

Cohee1207 commented:
Also, I wanted to make grammar a preset thing, not global.

Cohee1207 commented:
I kindly ask you to remove the `(ChatML)` part from the Mistral preset name. ChatML is a format to represent a JSON array of messages with role/name/content properties, nothing else. `<|im_start|>` and others are the internal format text prompt sequences that ChatML gets converted to when prompting the GPT-3.5 and GPT-4 models. Let's not cause confusion in the terminology usage.

WolframRavenwolf committed:
Renamed Mistral-OpenOrca (removed ChatML)

WolframRavenwolf commented:
> I kindly ask you to remove the `(ChatML)` part from the Mistral preset name. ChatML is a format to represent a JSON array of messages with role/name/content properties, nothing else. `<|im_start|>` and others are the internal format text prompt sequences that ChatML gets converted to when prompting the GPT-3.5 and GPT-4 models. Let's not cause confusion in the terminology usage.

Alright, agreed and done! (It's what [TheBloke named the prompt template](https://huggingface.co/TheBloke/Mistral-7B-OpenOrca-GGUF#prompt-template-chatml), but it's apparently a misunderstanding as the original model card only says: "We used OpenAI's Chat Markup Language (ChatML) format, with `<|im_start|>` and `<|im_end|>` tokens added to support this.")

WolframRavenwolf commented:
> Singleline seems like a legacy thing too, Koboldcpp might not support it at all. It's very easily emulated however by adding a newline as a custom stopping string on all backends, that's how it essentially works.

I'll implement the other stuff as we discussed, but what about the "Single-line mode" option? Should I remove it completely or keep it as part of the preset? I agree with your reasoning that it's better to implement it as a custom stopping string when needed, and if it's not even compatible with KoboldCpp, I'd rather remove than keep it to avoid confusion. (And I don't see a real use case for this as an option anyway, especially now that we also have grammar.)

Cohee1207 commented:
Keep the option for Singleline (some might still be using it, for whatever reason), but remove it from presets for the same reason as EOS - not overwriting the pick with the preset value.

WolframRavenwolf commented:
OK, will do it like that.

Question about Response Length and Context Size - those are currently still saved inside the presets (but no default preset includes them). Do you want to keep it like that or should I remove them as well from the presets by filtering them out in preset-manager.js?

Cohee1207 commented:
I'd like to leave it as is: do not include it in default presets, save it in custom presets.

WolframRavenwolf commented:
OK, almost done now:

![Single-line mode](https://github.com/SillyTavern/SillyTavern/assets/52386626/c536b309-6005-4c26-a12f-b8d6cdbdf3dc)

KoboldCpp reflects the setting in its debug output, but it has no effect - even when true, I got multi-line responses. So I wrote the help text accordingly.

***

Hopefully final question: Should we set Streaming ON by default? Wouldn't affect old users but new users would get streaming by default, which I consider a very useful feature, so best user experience out of the box. (Streaming was also recently improved further in KoboldCpp, which I'd expect to be used by more new users coming from the local Llama scene than KoboldAI).

Cohee1207 commented:
"Should we set Streaming ON by default" - I'd say NO for the time being, because it's only Koboldcpp.

WolframRavenwolf committed:
Removed single_line from kobold presets

WolframRavenwolf committed:
Removed obsolete use_stop_sequence setting

WolframRavenwolf committed:
Ban EOS Token off by default

WolframRavenwolf committed:
Split AI Resp. Conf. in global and preset-specific settings

WolframRavenwolf committed:
Resolve conflicts

WolframRavenwolf commented:
Alright, I'm done with the main stuff!

@Cohee1207 Can you solve the merge conflict? It's because of the changes to linebreaks: The way it was before my changes didn't work in Chrome and Firefox, with my changes it at least works in Firefox. Chrome still doesn't want to output a linebreak inside the pre/code tags.

Since this took longer than expected, and the PR is big enough as it is, I'll have to postpone the other stuff from my TODO list and handle that in another PR at a later time. I'll turn the more relevant BUGFIX items into actual bug reports, then maybe you or someone else can take a look and possibly fix them in the meantime.

***

One more thing: I just noticed the newly released [ehartford/dolphin-2.0-mistral-7b · Hugging Face](https://huggingface.co/ehartford/dolphin-2.0-mistral-7b) - ehartford writes: "Prompt format: This model (and all my future releases) use [ChatML](https://github.com/openai/openai-python/blob/main/chatml.md) prompt format."

Oh well, another one that calls it ChatML. Looks like that term will stick for the prompt format.

But I'd like to rework the new "Mistral-OpenOrca" prompt template to also work with "Dolphin-Mistral" and others by changing the system prompt to be more generic so we don't need a new preset for every new model (I'm sure we'll be flooded with Mistral-based models soon which also use this format). However, I don't know what we should call the template if you don't like the ChatML moniker. Any suggestions?

Cohee1207 commented:
> Can you solve the merge conflict? 

Ok, I'll do that and start testing.

> Oh well, another one that calls it ChatML. Looks like that term will stick for the prompt format.

Well, the name is still wrong but stuck with the community. I'd try chatting with the Mistral team about that when I get a minute, it was their idea after all.

Cohee1207 committed:
Merge branch 'staging' into assorted-fixes-and-improvements

Cohee1207 commented on public/index.html:
```diff
@@ -119,79 +119,10 @@
                 </div>
                 <div class="scrollableInner">
+                    <h3>
+                        <span data-i18n="[title]AI Response Configuration">AI Response Configuration</span>
```
It doesn't need the [title] in the i18n attribute, this notation is for "title" (on-hover text) translations only.

Cohee1207 commented on public/i18n.json:
```diff
@@ -620,6 +623,9 @@
         "Streaming": "ストリーミング",
         "Display the response bit by bit as it is generated.": "生成されると、レスポンスをビットごとに表示します。",
         "When this is off, responses will be displayed all at once when they are complete.": "これをオフにすると、レスポンスは完了時に一度にすべて表示されます。",
+        "Generate only one line per request (KoboldAI only, ignored by KoboldCpp).": "NEEDS TRANSLATION",
```
I'd say remove it from i18n files or make it English. NEEDS TRANSLATION makes the UI unusable because it's impossible to know what it is. I know it's in the Japanese translations for a while, but we didn't have a hero to fix it.

WolframRavenwolf replied:
Ah, OK, I thought that's the "official" marker for as of yet untranslated strings.

I translated them using Google Translate and checked that the inverse translation still fits. Looks good to me!

WolframRavenwolf committed:
Fix title

WolframRavenwolf committed:
Add translations for new help texts

WolframRavenwolf committed:
Merge branch 'assorted-fixes-and-improvements' of https://github.com/StefanDanielSchwarz/SillyTavern into assorted-fixes-and-improvements

WolframRavenwolf committed:
Fix i18n.json whitespace

WolframRavenwolf committed:
Make Mistral-OpenOrca system prompt more generic

WolframRavenwolf commented:
> > Oh well, another one that calls it ChatML. Looks like that term will stick for the prompt format.
> 
> Well, the name is still wrong but stuck with the community. I'd try chatting with the Mistral team about that when I get a minute, it was their idea after all.

I see us getting into the same situation as with LLaMA 33B being called 30B. While technically not correct, the majority is going to use it, and we'll either go along with it or use complicated constructs like 30B/33B to indicate we mean the same thing. 😐

And I still can't think of a better name than "ChatML" for this instruct preset instead of "Mistral-OpenOrca" or something silly like "Mistral-Dolphin+OpenOrca+otherChatMLderivatives"... 😉

***

I've renamed "Mistral-OpenOrca" to "ChatML" again for lack of a better name and because people ask "where is the ChatML option in Silly Tavern". So while "ChatML" may not be optimal, apparently it's what users look for, and it's still far better than "Mistral-OpenOrca" as the preset's name. But if you really hate it, you can change it, of course.

WolframRavenwolf committed:
Renamed "Mistral-OpenOrca" to "ChatML" again

Cohee1207 commented:
I know you mainly target KoboldAI, but the new UI layout looks a bit inconsistent across the APIs with the positioning of the checkboxes in relation to the preset selector. It is certainly going to be a habit-breaker. I propose to leave the response length, context size slider and context unlock tick under the "AI Response Configuration" title and move the rest under the preset selector to keep it consistent for all the API sources.

WolframRavenwolf commented:
Really? I was using the preset selector as a separator, everything above it is global and not (by default) part of a preset, and everything underneath is affected by preset selection. I also checked all the other APIs and how they look, didn't think it's inconsistent. So I'm still in favor of this new sorting, but you're the boss here, so ultimately it's your decision.

If we move it back below the selector, I think we should somehow make it clear which options are global. There's also the inconsistency of the sliders at the top being saved in a preset, that could also be confusing.

Maybe we should add a small help text just below "Click slider numbers to input manually." at the top that says something like "*: Global option, not saved in preset." and add asterisks to those options. Or maybe "Persistent" or "Independent" would be a better descriptor than "Global"?

***

Thought about it some more: Maybe the best way would actually be reverting my change and moving "Single-line mode" and "Ban EOS Token" down to where they used to be. Since they're both disabled by default anyway, I see no problem with that, and I'd actually let them be saved in presets again as they were initially so if someone actively changes them and saves a preset, their setting will be saved.

That would bring us back to my original intention: Just having "Ban EOS Token" OFF by default so users who don't know/understand the setting don't get fooled (just had another user on Reddit who complained about the model outputting nonsense after the message which was solved by turning off "Ban EOS Token"). With that implemented now, there's no more need to keep those settings outside of customer preset, IMHO.

Would also keep things consistent as then all settings below the preset selector will get saved into the user presets. No need for an additional help text at the top and less confusion with what gets saved inside custom presets and what doesn't.

In fact, I'm for saving Response Length and Context Size in presets again, too. The number of models with different max context sizes is increasing (LLaMA 2K, Llama 2 4K, Mistral 8K-32K, Code Llama 16K), and I find myself manually editing these boxes more often (including toggling the "Unlocked" checkbox), so being able to save those to presets would be quite helpful now. So I propose we continue to keep all of the "global" settings out of the default presets, but let users save them again when creating custom presets.

(Ideally there would be a way for end-users to define what they want to save into presets, maybe by being able to toggle checkboxes next to the values that control which settings get saved and which don't, but that would make the UI more complex. Not sure if that's what you want, and for now, I don't think that advanced level of customization is absolutely required.)

WolframRavenwolf committed:
More (UI) fixes and improvements

WolframRavenwolf commented:
OK, reworked some changes. The settings UI is back to its original look, but with improved help texts and labels, plus better consistency.

- Reverted some changes (crossed them out in the above changelog)
- Renamed Kobold's "Rep. Pen. Slope" to "Repetition Penalty Slope" and moved it directly beneath the other Rep. Pen. settings
- Renamed TextGen's "Encoder Rep. Pen." to "Encoder Repetition Penalty" for consistency (string lenght is fine)
- Unhid "Repetition Penalty Range" in Simple mode because it's highly relevant when Context Size (which is visible in Simple mode) is increased
- Removed "single_line" and "use_default_badwordsids" from Preset Manager filteredKeys again for consistency (and their defaults are fine now)

Also found some more little bugs which I added to the BUGFIX list above. If you want some of those fixed within this PR, we could do that, but I guess it's a big enough PR by now and in my opinion, we could just wrap this up and handle other stuff in another one as not to bloat this further.

@Cohee1207 When do you plan the next release? If it's a while off, I could start another PR for some of the open issues i found and listed above, unless you plan to handle them yourself or release soon.

Cohee1207 commented:
Thanks, I'll take a look at the updates. I want to release until the end of this weekend. It's a bit past due already.

Cohee1207 committed:
Sendas hint fixed

Cohee1207 committed:
Merge branch 'staging' into assorted-fixes-and-improvements

WolframRavenwolf commented:
I've got to ask about this one planned enhancement:

- Model detection on generation instead of connection when local (I often load a new model and the change is not reflected in the metadata when I forget to click Connect again)

It keeps tripping me up again and again as I forget to click Reconnect when changing models. SillyTavern doesn't notice the change and extra.model doesn't get updated.

Would it be a trivial change to call whatever function is updating extra.model upon API connect, also when starting inference, to make sure extra.model is always up to date? I looked but couldn't figure out how/where extra.model gets set.
